### ISSUE TITLE
stream callback not called when db is large (~2000 entries)

### ISSUE THREAD

**Issue by rawmean** [thisbejim/Pyrebase#287]
`db.child("root/child_1").stream(callback, None)` works perfectly when `child_1` contains only a few items (~200), but the callback is never called (not even once when the stream is invoked first) when we have ~2000 entries under `child_1`.

**Comment by rawmean** [comment]
ok, it turns out that when you have ~2000 entries in the database, it takes about 30 minutes to get the first callback. I doubt if this delay comes from firebase because `.get().val()` does not include such delay.
Is it possible to reduce this delay in Pyrebase?

**Comment by jphdotam** [comment]
Yep, having the same issue myself.

**Comment by jeffersonkr** [comment]
Did u guys found some solution for this issue ?

**Comment by jphdotam** [comment]
Nope, but interestingly if you use the package "firebase-admin" the exact same issue occurs, so I assume it's at the firebase end, not the implementation of pyrebase.

**Comment by jeffersonkr** [comment]
So.. posibilities of resolution should be filter the data, to get less data, or use another language with on() methods?

**Comment by jeffersonkr** [comment]
What is the sema object?
Can u tell me if i got the idea? Get the data manualy with get method then at callback function i can pass that data?
Woah its make a sense.. but actually stream method call Stream class with call SSEclient to retrieve data, so how did u managed to stop this? or giving the params it stop itself?

**Comment by rawmean** [comment]
> What is the sema object?
> Can u tell me if i got the idea? Get the data manualy with get method then at callback function can i pass that data?
> Woah its make a sense.. but actually stream method call Stream class with call SSEclient to retrieve data, so how did u managed to stop this trigger? or giving the params it stop itself?

seme is a semaphore object. Pyrebase is not thread-safe and my code is multi-threaded. You can ignore the `sema` stuff if your code is not multi-threaded.

I'm not stopping `stream` from doing it's job; I just get a snapshot manually at the beginning. I do rely on `stream` to send me snapshots when something changes in the database. This is just to avoid the large delay at the beginning when the database is large (~2000 entries or more).

**Comment by jeffersonkr** [comment]
i dont know what im doing wrong..
the stream still try to get a large data from firebase

`id_obra = sqlitedb.get_catraca(mac_address)[0][0]
    dados_com_token = {}
    dados = db.child('obras/{0}/colaboradores/'.format(id_obra)).get().val()
    for key, i in dados.items():
        try:
            if i['token'] == "":
                pass
            else:
                dados_com_token[key] = i
        except:
            pass

    dados_total = {'data': dados_com_token, 'event': 'put', 'path': ''}
    stream_colaboradores(dados_total)
    my_stream = db.child('obras/{0}/colaboradores/'.format(id_obra)).stream(stream_colaboradores, None)`

**Comment by jeffersonkr** [comment]
Hi, just passing here to advise you guys i found a solution. the SDK got now listener implementation, now when i try to listen a data with 3000 and nested..  this return in just 1.5 sec!!!!

`from firebase_admin import db
import firebase_admin
from firebase_admin import credentials

def callback(message):
    print(message.data)
    print(message.event_type)
    print(message.path)

credencial = credentials.Certificate('credencial.json')
firebase_admin.initialize_app(credencial, {'databaseURL': 'https://teste-e4334.firebaseio.com/', 'httpTimeout': 10})
a = db.reference('obras/211/colaboradores').listen(callback)`

**Comment by rawmean** [comment]
> Hi, just passing here to advise you guys i found a solution. the SDK got now listener implementation, now when i try to listen a data with 3000 and nested.. this return in just 1.5 min!!!!
> 
> ```
> from firebase_admin import db
> import firebase_admin
> from firebase_admin import credentials
> 
> def callback(message):
>     print(message.data)
>     print(message.event_type)
>     print(message.path)
> 
> credencial = credentials.Certificate('credencial.json')
> firebase_admin.initialize_app(credencial, {'databaseURL': 'https://teste-e4334.firebaseio.com/', 'httpTimeout': 10})
> a = db.reference('obras/211/colaboradores').listen(callback)
> ```

Thank you for sharing this. So, now that firebase has released an official python SDK, we don't need to use this report anymore. This is great news!

Many thanks to creators of this repo as it helped a lot of people to connect to firebase using python while firebase did not provide an SDK.
